### PR TITLE
PHP8: base64_decode(): Passing null is deprecated

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

- see https://github.com/OpenMage/magento-lts/pull/4243#discussion_r1802057305
- closes https://github.com/OpenMage/magento-lts/pull/4282

### CODE REVIEW

**Review by ADDISON74 — APPROVED** [review]

**Comment by fballiano** [comment]
why call prepareFilterString if the string is null?

in
```php
if (Mage::helper('sales')->isModuleOutputEnabled('Mage_Rss')) {
            $filterString = $this->getParam($this->getVarNameFilter());
            $filter = Mage::helper('adminhtml')->prepareFilterString($filterString);
            $storeId = array_key_exists('store_id', $filter) ? $filter['store_id'] : null;

            if (Mage::helper('rss')->isRssAdminOrderNewEnabled($storeId)) {
                $slug = $storeId ? '/store/' . $storeId : '';
                $this->addRssList('rss/order/new' . $slug, Mage::helper('sales')->__('New Order RSS'));
            }
        }
```

why call isRssAdminOrderNewEnabled if there's no $storeId?

**Comment by sreichel** [comment]
> why call prepareFilterString if the string is null?

Thanks. Changed it.

> why call isRssAdminOrderNewEnabled if there's no $storeId?

Order RSS is storeview based (now). Check for default setting.

File: app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php [hunk 1]
```diff
@@ -141,21 +141,38 @@ protected function _prepareColumns()
...
         if (Mage::helper('sales')->isModuleOutputEnabled('Mage_Rss')) {
-            $filterString = $this->getParam($this->getVarNameFilter());
-            $filter = Mage::helper('adminhtml')->prepareFilterString($filterString);
-            $storeId = array_key_exists('store_id', $filter) ? $filter['store_id'] : null;
+            $storeId = null;
+
+            $filterString = $this->getParam($this->getVarNameFilter(), '');
+            if ($filterString) {
+                $filter = Mage::helper('adminhtml')->prepareFilterString($filterString);
+                $storeId = array_key_exists('store_id', $filter) ? $filter['store_id'] : null;
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php:166]: I opt for readability and consistency:
```suggestion
                $storeId = $filter['store_id'] ?? null;
```
Is it better?

Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php]: Yes :)

**Review by kiatng — APPROVED** [review]

File: app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php [hunk 2]
```diff
@@ -141,21 +141,38 @@ protected function _prepareColumns()
...
-            $storeId = array_key_exists('store_id', $filter) ? $filter['store_id'] : null;
+            $storeId = null;
+
+            $filterString = $this->getParam($this->getVarNameFilter(), '');
+            if ($filterString) {
+                $filter = Mage::helper('adminhtml')->prepareFilterString($filterString);
+                $storeId = $filter['store_id'] ?? null;
+            }
 
             if (Mage::helper('rss')->isRssAdminOrderNewEnabled($storeId)) {
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php:169]:
```suggestion
            if ($storeId !== null && Mage::helper('rss')->isRssAdminOrderNewEnabled($storeId)) {
```

Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php:169]: This whould hide the RSS Link when no store is filtered.

Comment (fballiano) [reply app/code/core/Mage/Adminhtml/Block/Sales/Order/Grid.php:169]: ok. I can't clone the thread btw

**Review by ADDISON74 — APPROVED** [review]
It fixes the reported issues.

**Pull request merged**